v3/lints/rfc/lint_name_constraint_not_fqdn.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package rfc
+
+import (
+	"strings"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type nameConstraintNotFQDN struct{}
+
+func (l *nameConstraintNotFQDN) Initialize() error {
+	return nil
+}
+
+func (l *nameConstraintNotFQDN) CheckApplies(c *x509.Certificate) bool {
+	return util.IsExtInCert(c, util.NameConstOID)
+}
+
+func (l *nameConstraintNotFQDN) Execute(c *x509.Certificate) *lint.LintResult {
+
+	incorrectPermittedHosts := make([]string, 0)
+	incorrectExcludedHosts := make([]string, 0)
+	errString := ""
+
+	for _, subtreeString := range c.PermittedURIs {
+		isHostFQDN(subtreeString.Data, &incorrectPermittedHosts)
+	}
+	for _, subtreeString := range c.ExcludedURIs {
+		isHostFQDN(subtreeString.Data, &incorrectExcludedHosts)
+	}
+
+	if len(incorrectPermittedHosts) != 0 {
+		errString += returnErrorString(incorrectPermittedHosts, true)
+	}
+	if len(incorrectPermittedHosts) != 0 && len(incorrectExcludedHosts) != 0 {
+		errString += "; "
+	}
+	if len(incorrectExcludedHosts) != 0 {
+		errString += returnErrorString(incorrectExcludedHosts, false)
+	}
+
+	if len(errString) != 0 {
+		return &lint.LintResult{
+			Status:  lint.Error,
+			Details: errString,
+		}
+	}
+
+	return &lint.LintResult{Status: lint.Pass}
+}
+
+func isHostFQDN(host string, incorrectHosts *[]string) {
+
+	host = strings.TrimPrefix(host, ".")
+
+	if !util.IsFQDN(host) {
+		*incorrectHosts = append(*incorrectHosts, host)
+	}
+
+}
+
+func returnErrorString(incorrectHosts []string, isInclusion bool) string {
```

Comment:
bikeshed nit: I would probably name this function something like `buildErrorString`, I think the "return" part is implicit in the function definition saying it returns a string. Just a personal preference!